client/my-sites/checkout/checkout/secure-payment-form.jsx

```diff
@@ -233,10 +233,13 @@ export class SecurePaymentForm extends Component {
 		} );
 
 		if ( forcedAtomicProducts.length ) {
-			defer( () => {
-				// Until Atomic sites support being private / unlaunched, set them to public on upgrade
-				this.props.saveSiteSettings( selectedSiteId, { blog_public: 1 } );
-				debug( 'Setting site to public because it is an Atomic plan' );
+			await new Promise( resolve => {
+				defer( async () => {
+					// Until Atomic sites support being private / unlaunched, set them to public on upgrade
+					await this.props.saveSiteSettings( selectedSiteId, { blog_public: 1 } );
```

Comment:
We should probably wrap this in a try / catch.

At very least, we should `reject` so if this call fails for whatever reason, at least there's a chance the transaction goes through.

Even better would be to dispatch some failed action.

Reply:
I know I'm always antsy when I've submitted payment info & there's some partial failure making me re-submit... ("Will I get charged twice?")

Reply:
Agree, we don't want `submitTransaction` to be called if saving fails.

Would the `SITE_SETTINGS_SAVE_FAILURE` dispatched inside `saveSiteSettings()` be enough for flagging the error though?

Reply:
@jblz `maybeSetSiteToPublic` now throws when `saveSiteSettings()` throws. This means the transaction doesn't get submitted (and the exception just floats off into the ether like action handler exceptions normally do).

Seeing if I can see what `SITE_SETTINGS_SAVE_FAILURE` makes the UI look like, at least triggering it on my machine.